packages/merge-on-green/test/merge-on-green.ts

```diff
@@ -981,109 +929,268 @@ describe('merge-on-green', () => {
         });
 
         scopes.forEach(s => s.done());
-
-        assert(!addPRStub.called);
-
-        logger.info('stub called? ' + addPRStub.called);
+      });
+    });
+    describe('stubbing all GCP functions', () => {
```

Comment:
I wouldn't make the describe block describe "how you're going to write your tests" (_which is just an implementation detail), rather it should describe what you're testing.

If I'm following, we're moving away from `nock` based tests with these new tests towards using sinon more?

What I might do is go so far as splitting it across two test files:

* integration.js <-- integration(ish) tests with all our nock tests.
* unit.js <-- unit tests using sinon mianly for mocking.

This could be follow up work, not a blocker 👍

Reply:
I ended up dividing the functions into `merge-logic` vs. `merge-on-green` wrapper logic, which I think does a better job of describing the split (although, to your point, maybe it's worth adding more integration tests to have its own separate file)